astroid/transforms.py

```diff
@@ -3,10 +3,13 @@
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
 import collections
-from functools import lru_cache
+from typing import TYPE_CHECKING
 
 from astroid.context import _invalidate_cache
 
+if TYPE_CHECKING:
+    from astroid.nodes.node_ng import NodeNG
```

Comment:
I know we have different opinions about this, but I like importing `nodes` here as it also keeps the line much shorter and gives some consistency with the `pylint` codebase. I'll leave this to you!